src/pages/SignUp.jsx

```diff
@@ -1,5 +1,169 @@
-import React from "react";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { Card, CardBody, Input, Link } from "@nextui-org/react";
+import InputForm from "../components/InputForm";
+import PasswordEye from "../components/PasswordEye";
+import LoadingButton from "../components/LoadingButton";
+import { register } from "../utils/authHandler";
 
-export default function SignUp() {
-  return <div>SignUp</div>;
+export default function Login() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [emailError, setEmailError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+  const [confirmPasswordError, setConfirmPasswordError] = useState("");
+  const [loginStatus, setLoginStatus] = useState({
+    success: null,
+    message: null,
+  });
+  const [showPassword, setShowPassword] = useState(false);
+  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const navigate = useNavigate();
+
+  function updateEmail(value) {
+    setEmail(value);
+    setEmailError("");
+  }
+
+  function updatePassword(value) {
+    setPassword(value);
+    setPasswordError("");
+  }
+
+  function updateConfirmPassword(value) {
+    setConfirmPassword(value);
+    setConfirmPasswordError("");
+  }
+
+  function isInvalidEmail(email) {
+    if (!email) return "Enter Email";
+    if (!email.match(/^[A-Z0-9._%+-]+@[A-Z0-9.-]+.[A-Z]{2,4}$/i))
+      return "Invalid Email";
+    return false;
+  }
+
+  function isInvalidPassword(password) {
+    const minLength = 8
+    const excludeList = ["password", "123", "1234", "12345", "123456"];
+    const specialCharacters = "!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~";
+    const lowercase = password.toLowerCase();
+    if (!password) return "Enter Password";
```

Comment:
(comment): Although it may be personal preference, you should be aware that many teams require devs to use curly braces for all conditional statements, even simple statements like this. Feel free to omit the curly braces if you want.